pentaho-js/src/main/javascript/cdf/components/TableComponent.js

```diff
@@ -600,6 +600,7 @@ define([
             target = target.closest('td');
           }
           var position = myself.dataTable.fnGetPosition(target.get(0));
+          if (position == null) { return; }
```

Comment:
Alternatively you can do some validations before initializing the position variable, such as:
```
if(!myself.dataTable.api(true).cell( target.get(0) ).index()) {
  console.warn("Click on invalid data detected");
  return;
}
var position = myself.dataTable.fnGetPosition(target.get(0));
```